src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -204,14 +206,18 @@ export class TerminalService implements ITerminalService {
 				: Promise.resolve();
 		this._offProcessTerminalService = !!this._environmentService.remoteAuthority ? this._remoteTerminalService : this._localTerminalService;
 		initPromise.then(() => this._setConnected());
-
-		// Wait up to 5 seconds for profiles to be ready so it's assured that we know the actual
+		this._initRequiresWindowsMode();
+    	// Wait up to 5 seconds for profiles to be ready so it's assured that we know the actual
 		// default terminal before launching the first terminal. This isn't expected to ever take
 		// this long.
 		this._profilesReadyBarrier = new AutoOpenBarrier(5000);
 		this._refreshAvailableProfiles();
 	}
 
+	private async _initRequiresWindowsMode(): Promise<void> {
+		this._requiresWindowsMode = await this._offProcessTerminalService?.requiresWindowsMode() || false;
+	}
```

Comment:
I think we'd be better off evaluating this every time when the process gets created. Right now this wouldn't work if you toggled conpty/winpty and would also cause issues when creating local terminals in remote workspaces.

I mentioned the traits idea yesterday but wasn't too keen on requiring a round trip to access it, here's a better idea:

```ts
IPtyService.createProcess(...): Promise<{
  persistentProcessId: number,
  requiresWindowsMode?: boolean
}>
```